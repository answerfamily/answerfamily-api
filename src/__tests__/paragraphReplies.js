const {
  loadESFixtures,
  unloadESFixtures,
  loadMongoFixtures,
  unloadMongoFixtures,
} = require('../../test/fixtures');
const gql = require('../../test/gql');
const {
  ES_FIXTURES,
  MONGO_FIXTURES,
} = require('../__fixtures__/articleParagraphReply');

describe('paragraphReplies', () => {
  beforeAll(async () => {
    await loadESFixtures(ES_FIXTURES);
    await loadMongoFixtures(MONGO_FIXTURES);
  });

  afterAll(async () => {
    await unloadESFixtures(ES_FIXTURES);
    await unloadMongoFixtures(MONGO_FIXTURES);
  });

  it('lists unfiltered paragraphReplies', async () => {
    const { data, errors } = await gql`
      {
        paragraphReplies {
          paragraph {
            id
          }
          reply {
            id
          }
        }
      }
    `();

    expect(errors).toBeUndefined();
    expect(data).toMatchSnapshot();
  });

  it('does limiting', async () => {
    const { data, errors } = await gql`
      {
        paragraphReplies(first: 1) {
          paragraph {
            id
          }
          reply {
            id
          }
        }
      }
    `();

    expect(errors).toBeUndefined();
    expect(data).toMatchSnapshot();
  });

  it('does skipping', async () => {
    const { data, errors } = await gql`
      {
        paragraphReplies(skip: 1) {
          paragraph {
            id
          }
          reply {
            id
          }
        }
      }
    `();

    expect(errors).toBeUndefined();
    expect(data).toMatchSnapshot();
  });
});
