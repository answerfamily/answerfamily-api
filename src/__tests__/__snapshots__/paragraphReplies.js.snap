// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`paragraphReplies does limiting 1`] = `
Object {
  "paragraphReplies": Array [
    Object {
      "paragraph": Object {
        "id": "p2",
      },
      "reply": Object {
        "id": "r2",
      },
    },
  ],
}
`;

exports[`paragraphReplies does skipping 1`] = `
Object {
  "paragraphReplies": Array [
    Object {
      "paragraph": Object {
        "id": "p1",
      },
      "reply": Object {
        "id": "r1",
      },
    },
  ],
}
`;

exports[`paragraphReplies lists unfiltered paragraphReplies 1`] = `
Object {
  "paragraphReplies": Array [
    Object {
      "id": "5bd677ef092a3ee2541b1a1d",
      "paragraph": Object {
        "id": "p2",
      },
      "reply": Object {
        "id": "r2",
      },
    },
    Object {
      "id": "5bd677bb092a3ee2541b1a0d",
      "paragraph": Object {
        "id": "p1",
      },
      "reply": Object {
        "id": "r1",
      },
    },
  ],
}
`;
